### PR TITLE
feature(aion): Create room form

### PR DESCRIPTION
**Pull request by jtkpiotr**
**Summary**
screen shot 2017-09-07 at 22 01 21

**Related issues**

**Test plan**

### CODE REVIEW

File: aion/web/elm/elm-package.json
Comment (pmrukot) [inline aion/web/elm/elm-package.json]: Unnecessary empty line.

File: aion/web/elm/src/App.elm
Comment (pmrukot) [inline aion/web/elm/src/App.elm]: Move import to match alphabetical order.

File: aion/web/elm/src/Msgs.elm
Comment (pmrukot) [inline aion/web/elm/src/Msgs.elm]: Move import to match alphabetical order.

File: aion/web/elm/src/Panel/Models.elm
Comment (pmrukot) [inline aion/web/elm/src/Panel/Models.elm]: `name` property should be in a new line.

File: aion/web/elm/src/Panel/View.elm
Comment (pmrukot) [inline aion/web/elm/src/Panel/View.elm]: Move import to match alphabetical order.

File: aion/web/elm/src/Update.elm
Comment (pmrukot) [inline aion/web/elm/src/Update.elm]: Why would you name it `oldPanelData` while you are not updating it anyway? Also I would rename `roomForm` to distinguish variable from actual record.

**Review by pmrukot — COMMENTED** [review]
Looks solid, added some insights. Make sure to run elm-format and credo so that travis build passes. Also some screenshots would be nice as a proof.

**Comment by mrapacz** [comment]
@jtkpiotr a few notes for the future:
1. Run make install-hooks to install our pre-commit hook, you will not forget about running elm-format / credo then.

2. When you open pull request stay consistent with our naming convention, use PR titles and merge commit messages that match this regex: ```(feature|fix|enhancement)\([a-z]+\):\ [A-Z].+``` Also please fill in the prepared sections of the PR message. If there are issues that are connected (and there should always be) you wanna make sure that we know what exactly is the problem you tried to solve. We also wanna know how we can test your changes and how you know that they are working + a few words on what's happening in the code / basically anything that is needed to give us a vision of your work.

3. When you manage your PR you can add some labels we have prepared just so we know what's the current state of your work. There are many PRs that are created but don't need review because they are still in development process. If you just create the PR but it's not the final form, you can add the ``Work in progress`` label, if you've finished your work and want us to review your code you can add the ``Waiting for review`` label + assign it to us (this way we will get notifications and we'll know that it's the final form of your work. 

4. Protip: if you write ```closes: #issue_number``` in the ```Related issues``` section, after merging your PR github will automagically close the given issue.

File: aion/web/elm/src/Panel/View.elm
Comment (mrapacz) [inline aion/web/elm/src/Panel/View.elm:139]: I made a special issue for correcting things like `bellow`, but it can be fixed before merging anyway #110

File: aion/web/elm/src/Update.elm
Comment (mrapacz) [inline aion/web/elm/src/Update.elm]: I guess we could get a better naming than x. The moment somebody wants to change the logic here, they'll have to log what the x is. Also, do we still use the name 'subject'?

File: aion/web/elm/src/Update.elm
Comment (mrapacz) [inline aion/web/elm/src/Update.elm]: What's the reason for the sub prefix in subMsg subCmd and subModel? I'm curious, perhaps there is a better name for that.

File: aion/web/elm/src/Update.elm
Comment (mrapacz) [inline aion/web/elm/src/Update.elm]: I think we can use the let part to actually assign the final value to a name like updatedCategoryMultiSelect (or whatever seems more suitable for you) and then update the model like:
```
{model | panelData = {oldPanelData | categoryMultiSelect = updatedCategoryMultiSelect}} 

```
Having function calls in the record updates makes it harder to read the code.

File: aion/test/controllers/room_controller_test.exs
Comment (mrapacz) [inline aion/test/controllers/room_controller_test.exs]: Kudos for testing stuff :+1:

**Review by mrapacz — CHANGES_REQUESTED** [review]

File: aion/web/elm/src/Update.elm
Comment (pmrukot) [reply aion/web/elm/src/Update.elm]: I think it is because its subscription related update.

**Comment by pmrukot** [comment]
@mrapacz PR has been adjusted and is ready to be merged.

**Review by pmrukot — APPROVED** [review]

File: aion/web/elm/elm-package.json
Comment (mrapacz) [inline aion/web/elm/elm-package.json:23]: The fact that there are no trailing commas allowed in elixir really grinds my gears.

**Review by mrapacz — APPROVED** [review]
XP at its finest

File: aion/web/elm/elm-package.json
Comment (pmrukot) [reply aion/web/elm/elm-package.json:23]: Its not elixir, thats just JSON format mate :)

Comment (mrapacz) [reply aion/web/elm/elm-package.json:23]: Yeah right, sorry, that's just late night me. Anyway this is also what I experienced while swapping deps in mix.exs a lot lately, so seeing the deps list, I didn't hesitate before blaming elixir 😆

**Pull request merged**
